public/nextjs_migration/client/js/addEmail.js

```diff
@@ -0,0 +1,74 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+let dialogEl, form
+
+function init () {
+  dialogEl = document.querySelector('dialog[data-partial="addEmail"]')
+  if (!dialogEl) return
+
+  form = dialogEl.querySelector('form')
+  form.addEventListener('submit', handleSubmit)
+  dialogEl.addEventListener('close', kill)
+}
+
+async function handleSubmit (e) {
+  e.preventDefault()
+
+  try {
+    form.elements['email-submit'].toggleAttribute('disabled', true)
+    const res = await fetch('/api/v1/user/email', {
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'text/html' // set to request localized response
+      },
+      mode: 'same-origin',
+      method: 'POST',
+      body: JSON.stringify({
+        email: form.elements['email-address'].value
+      })
+    })
+
+    if (!res.ok) throw new Error()
+
+    window.gtag('event', 'added_email', { result: 'success' })
+
+    const { newEmailCount } = await res.json()
+
+    renderSuccess({
+      email: form.elements['email-address'].value,
+      newEmailCount
+    })
+  } catch (e) {
+    // TODO: localize error messages
+    const toast = document.createElement('toast-alert')
+    toast.textContent = `Could not add email. ${e.message}`
```

Comment:
Yeah it's super dirty, as it could be anything that errors thrown in the `try {}` block set it to - I've even seen it to be set to `document.querySelector(...) is null` when I hadn't properly ported that part yet :disappointed: